src/org/sosy_lab/java_smt/basicimpl/withAssumptionsWrapper/BasicProverWithAssumptionsWrapper.java

```diff
@@ -36,6 +37,21 @@ protected void clearAssumptions() {
     solverAssumptionsAsFormula.clear();
   }
 
+  protected void clearAssumptionsWithInterruptedException() throws InterruptedException {
+    for (int i = 0; i < solverAssumptionsAsFormula.size(); i++) {
+      try {
+        delegate.pop();
+      } catch (IllegalStateException ise) {
+        if (ise.getMessage().startsWith(AbstractProver.SHUTDOWN_EXCEPTION_PREFIX)) {
+          throw new InterruptedException(
+              ise.getMessage()
+                  .replace("Prover is not usable due " + "to shutdown with message: ", ""));
+        }
+      }
```

Comment:
Why so complicated and even using String matching?

The catch block could simply call `shutdownIfNecessary()`, for example. Or the `IllegalStateException` could have an `InterruptedException` as cause.

Also, there is a bug here: If there is an `IllegalStateException` for a different reason, it is swallowed!

Reply:
> Because they are placed in the `AbstractProver`, and this wrapper is not inheriting from it. The delegate might be some other wrapper that also doesn't.

Why shouldn't `BasicProverWithAssumptionsWrapper` get the shutdown notifier? It can simply be passed in as constructor parameter.

> We could of course try to access the nested delegates until we get to where it is accessible in all wrappers.

I don't understand this sentence, sorry.